pymeasure/instruments/activetechnologies/AWG401x.py

```diff
@@ -0,0 +1,1145 @@
+          circularly.
+        * STEP<PED>: the AWG, for each entry, waits for a trigger event before
+          the execution of the sequencer entry. The waveform of the entry will
+          loop as written in the entry repetition parameter. After the
+          generation of an entry has completed, the last sample of the current
+          entry or the first sample of the next entry is held until the next
+          trigger is received. At the end of the entire sequence the execution
+          will restart from the first entry.
+        * ADVA<NCED>: it enables the “Advanced” mode. In this mode the
+          execution of the sequence can be changed by using conditional and
+          unconditional jumps (JUMPTO and GOTO commands) and dynamic jumps
+          (PATTERN JUMP commands).
+
+        The \\*RST command sets this parameter to CONTinuous.""",
+        validator=strict_discrete_set,
+        values=["CONTINUOUS", "CONT", "BURST", "BURS", "TCONTINUOUS",
+                "TCON", "STEPPED", "STEP", "ADVANCED", "ADVA"]
+    )
+
+    burst_count = Instrument.control(
+        "AWGControl:BURST?",
+        "AWGControl:BURST %d",
+        """This property sets or queries the burst count parameter.""",
+        validator=strict_range,
+        dynamic=True
+    )
+
+    burst_count_max = Instrument.measurement(
+        "AWGControl:BURST? MAXimum",
+        """This property queries the maximum burst count parameter."""
+    )
+
+    burst_count_min = Instrument.measurement(
+        "AWGControl:BURST? MINimum",
+        """This property queries the minimum burst count parameter."""
+    )
+
+    sampling_rate = Instrument.control(
+        "AWGControl:SRATe?",
+        "AWGControl:SRATe %f",
+        """This property sets or queries the sample rate for the Sampling
+        Clock.""",
+        validator=strict_range,
+        dynamic=True
+    )
+
+    sampling_rate_max = Instrument.measurement(
+        "AWGControl:SRATe? MAXimum",
+        """This property queries the maximum sample rate for the Sampling
+        Clock."""
+    )
+
+    sampling_rate_min = Instrument.measurement(
+        "AWGControl:SRATe? MINimum",
+        """This property queries the minimum sample rate for the Sampling
+        Clock."""
+    )
+
+    _run_state_dict = {0: "STOPPED", 1: "WAITING_TRIGGER", 2: "RUNNING"}
+
+    run_status = Instrument.measurement(
+        "AWGControl:RSTATe?",
+        """This property returns the run state of the AWG. The possible values
+        are: STOPPED, WAITING_TRIGGER, RUNNING""",
+        get_process=lambda v: AWG401x_AWG._run_state_dict[v]
+    )
+
+    enabled = Instrument.control(
+        "AWGControl:RSTATe?", "AWGControl:%s",
+        """A boolean property that enables the generation of signals.""",
+        validator=strict_discrete_set,
+        values={True: "RUN", False: "STOP"},
+        map_values=True,
+        get_process=lambda v: "STOP" if v == 0 else "RUN"
```

Comment:
This property has different commands to set and get. Is this intentional?

Reply:
Yes, it is.  Here there are tree different command for a single property:
 * AWGControl: RSTATe? (Query Only)
 * AWGControl:RUN (No Query Form)
 * AWGControl:STOP (No Query Form)